imap_processing/spice/geometry.py

```diff
@@ -149,13 +149,45 @@ def get_spacecraft_spin_phase(
     spin_phase : float or np.ndarray
         Spin phase for the input query times.
     """
+    spin_df = get_spin_data()
+
+    # Combine spin_start_sec and spin_start_subsec to get the spin start
+    # time in seconds. The spin start subseconds are in milliseconds.
+    # TODO: Decide if we should do this calculation in the data itself
+    # or in get_spin_data function.
+    spin_df["spin_start_time"] = (
+        spin_df["spin_start_sec"] + spin_df["spin_start_subsec"] / 1e3
+    )
+
     if isinstance(query_met_times, float):
```

Comment:
This is the same code duplicated in both paths essentially. You should be able to do something like:
1. see what we got as input `is_scalar = isinstance(query_met_times, numbers.Number)`
2. convert met_times to an array regardless `query_met_times = np.asarray(query_met_times)` and use that for the calculations
3. special-case the return value `if is_scalar: convert_to_scalar, ...`